flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCUpsertOutputFormat.java

```diff
@@ -116,13 +116,15 @@ public void open(int taskNumber, int numTasks) throws IOException {
 			this.scheduler = Executors.newScheduledThreadPool(
 					1, new ExecutorThreadFactory("jdbc-upsert-output-format"));
 			this.scheduledFuture = this.scheduler.scheduleWithFixedDelay(() -> {
-				if (closed) {
-					return;
-				}
-				try {
-					flush();
-				} catch (Exception e) {
-					flushException = e;
+				synchronized (JDBCUpsertOutputFormat.this) {
```

Comment:
Why should we add `synchronized`? `flush()` is already `synchronized`, right?

Reply:
Make sure the flusher will graceful exit after close.